pkg/deployments/tasks/20220628-optimism-root-gauge-factory/test/task.fork.ts

```diff
@@ -0,0 +1,240 @@
+    await gaugeController.connect(veBALHolder).vote_for_gauge_weights(gauge.address, 10000); // Max voting power is 10k points
+
+    // We now need to go through an epoch for the votes to be locked in
+    await advanceTime(DAY * 8);
+
+    await gaugeController.checkpoint();
+    expect(await gaugeController['gauge_relative_weight(address)'](gauge.address)).to.be.gt(0);
+  });
+
+  it('mint & bridge tokens', async () => {
+    // The gauge has votes for this week, and it will mint the first batch of tokens. We store the current gauge
+    // relative weight, as it will change as time goes by due to vote decay.
+    const firstMintWeekTimestamp = await currentWeekTimestamp();
+    const gaugeRelativeWeight = await gaugeController['gauge_relative_weight(address)'](gauge.address);
+
+    const calldata = gauge.interface.encodeFunctionData('checkpoint');
+
+    // Even though the gauge has relative weight, it cannot mint yet as it needs for the epoch to finish
+    const zeroMintTx = await authorizerAdaptor.connect(admin).performAction(gauge.address, calldata);
+    expectEvent.inIndirectReceipt(await zeroMintTx.wait(), gauge.interface, 'Checkpoint', {
+      periodTime: firstMintWeekTimestamp.sub(WEEK), // Process past week, which had zero votes
+      periodEmissions: 0,
+    });
+    // No token transfers are performed if the emissions are zero, but we can't test for a lack of those
+
+    await advanceTime(WEEK);
+
+    // The gauge should now mint and send all minted tokens to the Arbitrum bridge
+    const mintTx = await authorizerAdaptor.connect(admin).performAction(gauge.address, calldata);
+    const event = expectEvent.inIndirectReceipt(await mintTx.wait(), gauge.interface, 'Checkpoint', {
+      periodTime: firstMintWeekTimestamp,
+    });
+    const actualEmissions = event.args.periodEmissions;
+
+    // The amount of tokens minted should equal the weekly emissions rate times the relative weight of the gauge
+    const weeklyRate = (await BALTokenAdmin.getInflationRate()).mul(WEEK);
+
+    const expectedEmissions = gaugeRelativeWeight.mul(weeklyRate).div(FP_SCALING_FACTOR);
+    expectEqualWithError(actualEmissions, expectedEmissions, 0.001);
+
+    // Tokens are minted for the gauge
+    expectTransferEvent(
+      await mintTx.wait(),
+      {
+        from: ZERO_ADDRESS,
+        to: gauge.address,
+        value: actualEmissions,
+      },
+      BAL
+    );
+
+    // And the gauge then deposits those in the predicate via the bridge mechanism
+    const bridgeInterface = new ethers.utils.Interface([
+      'event ERC20DepositInitiated(address indexed _l1Token, address indexed _l2Token, address indexed _from, address _to, uint256 _amount, bytes _data)',
+    ]);
+
+    expectEvent.inIndirectReceipt(await mintTx.wait(), bridgeInterface, 'ERC20DepositInitiated', {
+      _l1Token: BAL,
+      _l2Token: OptimismBAL,
+      _from: gauge.address,
+      _to: recipient.address,
+      _amount: actualEmissions,
+    });
```

Comment:
Can you manually make a call from an EOA to the bridge to the same function (approval + bridge) and link the L1 and L2 txs (the receipt of tokens on the L2) in this PR, so we can double-check that this event does mean the tokens will be received as expected?

Reply:
You intereacted with address 0x99c9fc46f92e8a1c0dec1b1747d010903e884be1 there, not the one on the input file. Where does this difference come from?

Reply:
Are you getting confused with the `OptimismBAL` address? `0x99C9fc46f92E8a1c0deC1b1747d010903E884bE1` is the address of the bridge in the `input.ts` file.